src/sysctl/sysctl.c

```diff
@@ -106,11 +106,12 @@ static int parse_file(OrderedHashmap *sysctl_options, const char *path, bool ign
 
         log_debug("Parsing %s", path);
         for (;;) {
-                char l[LINE_MAX], *p, *value, *new_value, *property, *existing;
+                char *p, *value, *new_value, *property, *existing;
+                _cleanup_free_ char *l = NULL;
                 void *v;
                 int k;
 
-                if (!fgets(l, sizeof(l), f)) {
+                if (!read_line(f, LONG_LINE_MAX, &l)) {
```

Comment:
`read_line` returns 0 on EOF, >0 on success, and <0 on error. See https://github.com/systemd/systemd/blob/master/src/basic/hostname-util.c#L234 for an example of how to handle those return values.

Reply:
uh, how did i miss that...